app/controllers/topological_inventory/ingress_api/v0/inventory_controller.rb

```diff
@@ -1,44 +1,34 @@
+require "topological_inventory/ingress_api/messaging_client"
+
 module TopologicalInventory
   module IngressApi
     module V0
       class InventoryController < ApplicationController
         skip_before_action :verify_authenticity_token
         def save_inventory
-          messaging_client.publish_message(
-            :service => "platform.topological-inventory.persister",
-            :message => "save_inventory",
-            :payload => JSON.parse(request.body.read),
-          )
+          payload = JSON.parse(request.body.read)
+
+          TopologicalInventory::IngressApi.with_messaging_client do |client|
+            client.publish_message(save_inventory_payload(payload))
+          end
 
           render status: 200, :json => {
             :message => "ok",
           }.to_json
-        rescue Kafka::MessageSizeTooLarge => e
-          render :status => 500, :json => {
-            :message    => e.message,
-            :error_code => e.class.to_s,
-          }.to_json
         rescue => e
           render :status => 500, :json => {
             :message    => e.message,
             :error_code => e.class.to_s,
           }.to_json
-
-          raise e
         end
 
         private
 
-        def messaging_client
-          ManageIQ::Messaging::Client.open(messaging_opts)
-        end
-
-        def messaging_opts
+        def save_inventory_payload(raw_payload)
           {
-            :protocol => :Kafka,
-            :host     => ENV["QUEUE_HOST"] || "localhost",
-            :port     => ENV["QUEUE_PORT"] || "9092",
-            :encoding => "json",
+            :service => "platform.topological-inventory.persister",
+            :message => "save_inventory",
+            :payload => raw_payload,
```

Comment:
i think the `:encoding => "json"` must stay? Otherwise it will start to send there yaml

Reply:
ah, you moved it